pinot-core/src/main/java/org/apache/pinot/core/indexsegment/generator/SegmentGeneratorConfig.java

```diff
@@ -469,6 +459,28 @@ public Schema getSchema() {
   public void setSchema(Schema schema) {
     Preconditions.checkNotNull(schema);
     _schema = schema;
+
+    // Set time related fields
```

Comment:
RealtimeSegmentConverter uses SegmentGeneratorConfig and sets time unit and time column name independently after constructing it with schema. Perhaps you can remove those lines form there? And move the TODOs here?

Reply:
Done